hrjobcontract/js/test/controllers/modal/modal-contract_test.js

```diff
@@ -1,45 +1,102 @@
-      _ContractDetailsService_) {
+    beforeEach(module(function ($provide) {
+      $provide.factory('ContractHealthService', function () {
+        return {
+          getOptions: function () {},
+          getFields: jasmine.createSpy(),
+          save: jasmine.createSpy()
+        }
+      });
+    }));
+
+    beforeEach(inject(function (_$controller_, _$rootScope_, _$httpBackend_, _$q_,
+      _ContractDetailsService_, _ContractHealthService_) {
       $controller = _$controller_;
       $rootScope = _$rootScope_;
       $httpBackend = _$httpBackend_;
       ContractDetailsService = _ContractDetailsService_;
+      ContractHealthService = _ContractHealthService_;
       $q = _$q_;
     }));
 
-    beforeEach(function() {
+    beforeEach(function () {
       $httpBackend.whenGET(/action=validatedates&entity=HRJobDetails/).respond({ success: true });
       $httpBackend.whenGET(/action=get&entity=HRJobContract/).respond({});
       $httpBackend.whenPOST(/action=create&entity/).respond({});
       $httpBackend.whenPOST(/action=replace&entity/).respond({});
+      $httpBackend.whenGET(/action=get&entity=HRHoursLocation/).respond({});
+      $httpBackend.whenGET(/action=get&entity=HRPayScale/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobDetails/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobHour/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobPay/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobLeave/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobHealth/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobPension/).respond({});
       $httpBackend.whenGET(/action=getoptions&entity=HRJobHealth/).respond({});
       $httpBackend.whenGET(/views.*/).respond({});
     });
 
-    beforeEach(function() {
+    beforeEach(function () {
+      var health = {};
+
+      $rootScope.$digest();
+      health.plan_type = {};
+      health.plan_type_life_insurance = {};
+      $rootScope.options = {
+        health: health
+      };
+    });
+
+    beforeEach(function () {
       mockUIBModalInstance();
       mockUIBModal('edit');
-      mockObjects();
+      contractHealthServiceSpy();
       makeController();
       createContractDetailsServiceSpy(true);
       createUIBModalSpy();
     });
 
+    describe("init()", function () {
+      describe("when ModalContractCtrl is initialized", function () {
+
+        beforeEach(function () {
+          $rootScope.$digest();
+        });
+
+        var result = {
+          Family: "Family",
+          Individual: "Individual"
+        };
+
+        it("calls getOptions() form ContractHealthService", function () {
+          expect(ContractHealthService.getOptions).toHaveBeenCalled();
+        });
+
+        it("fetches health insurance plan types", function () {
+          expect($rootScope.options.health.plan_type).toEqual(result)
```

Comment:
same question as before, how do you know that it's going to be equal to `result`?